### DISCUSSION TITLE
Write output into individual files

### DISCUSSION THREAD

**Discussion by Marvin-D-Martian** [ImageMagick/ImageMagick#6303] [Help] [answered]
### ImageMagick version

7.1.1.1-8

### Operating system, version and so on

Win 11 Pro

### Description

HI all. I need help. I googled and bing-ed and still..
I am going crazy. I would like to think, this should be a simple thing, but I guess I am just thick...

I am running:
`composite -gravity center X:\DIR01\*.png X:\DIR02\*.png X:\OUTPUT\%d.png`

Which saves all files in both directories into one file called 0.png...  I can't find a working example anywhere...

For the life of me, how do I process all (sequentially named) files in DIR01 and overlay them with files from DIR02 (matching sequential filenames) and write the result to **individual files**?

Thanks in advance

**Comment by snibgo** [comment, marked as answer]
I suggest using a shell script. This will loop through all the filenames in one directory, and call a `magick` command for each pair of files.

If the files are small enough to fit into memory at the same time, you could use "`-layers composite`" on the two sets of files.

I suggest using `magick`, not `composite`.

**Reply by Marvin-D-Martian** [reply]
Thanks for the swift reply. I was already thinking about using a script, but wanted to make sure I am not missing anything as I am quite new to magick. Thanks for the advice and your time!!

**Comment by snibgo** [comment]
You could do this:
```
magick X:\DIR01\*.png NULL: X:\DIR02\*.png -gravity center -compose Over -layers composite X:\OUTPUT\%d.png
```
But it will fail or be very slow if you don't have enough memory for all the input images.

**Reply by Marvin-D-Martian** [reply]
Thanks. I had luck with the script and it worked a charm. I will save this command for later work. It's definitely simpler than scripting it. I have 48GB RAM, should be just fine for what I want to do. Thanks again. Really appreciate your time.
